### PR TITLE
158039179 selection lib

### PR DESCRIPTION
**Pull request by mjbradvica**
Selection library models and validation.

### CODE REVIEW

File: src/Housing.Selection.Library/Housing.Selection.Library.csproj [hunk 1]
```diff
@@ -5,4 +5,11 @@
     <TargetFramework>netstandard2.0</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Reference Include="System.ComponentModel.Annotations">
+      <HintPath>..\..\..\..\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.netcore.app\2.0.0\ref\netcoreapp2.0\System.ComponentModel.Annotations.dll</HintPath>
+    </Reference>
```
Comment (escalonn) [inline src/Housing.Selection.Library/Housing.Selection.Library.csproj]: delete this reference, & re-add using correct nuget package providing this assembly

File: src/Housing.Selection.Library/Name.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Housing.Selection.Library
+{ 
+   public  class Name
```
Comment (escalonn) [inline src/Housing.Selection.Library/Name.cs]: format issue

File: src/Housing.Selection.Library/Batch.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+        public string BatchName { get; set; }
+
+        public int BatchOccupancy { get; set; }
+
+        public string BatchSkill { get; set; }
+
+        public ICollection<User> Users { get; set; }
+
+        public Address Address { get; set; }
```
Comment (escalonn) [inline src/Housing.Selection.Library/Batch.cs]: batch has location, not address. (likewise `Address.Batches`)

**Comment by mjbradvica** [comment]
Changes have been fixed.

File: src/Housing.Selection.Library/User.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
```
Comment (escalonn) [inline src/Housing.Selection.Library/User.cs]: since we don't need dataannotations remove the reference from project

**Comment by mjbradvica** [comment]
We need those for EF to enable cascade delete/update.

**Comment by escalonn** [comment]
still looking for change to make batch have state

**Comment by mjbradvica** [comment]
Fixed.

File: src/Housing.Selection.Library/Address.cs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+        public string City { get; set; }
+
+        public string State { get; set; }
+
+        public string PostalCode { get; set; }
+
+        public string Country { get; set; }
+
+        public ICollection<Batch> Batches { get; set; }
```
Comment (escalonn) [inline src/Housing.Selection.Library/Address.cs]: remove batches here

File: src/Housing.Selection.Library/Batch.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+        public DateTime EndDate { get; set; }
+
+        public string BatchName { get; set; }
+
+        public int BatchOccupancy { get; set; }
+
+        public string BatchSkill { get; set; }
+
+        public string Location { get; set; }
```
Comment (escalonn) [inline src/Housing.Selection.Library/Batch.cs]: remove location here

**Comment by mjbradvica** [comment]
fixed

**Pull request merged**